src/server/pkg/hashtree/error.go

```diff
@@ -0,0 +1,37 @@
+package hashtree
+
+import "fmt"
+
+// hashTreeError is a custom error type, returned by HashTree's methods
+type hashTreeError struct {
+	code ErrCode
+	s    string
+}
+
+func (e *hashTreeError) Error() string {
+	return e.s
+}
+
+// Code returns the "error code"  of 'err' if it was returned by one of the
+// HashTree methods, or "Unknown" if 'err' was emitted by some other function
+// (error codes are defined in interface.go)
+func Code(err error) ErrCode {
```

Comment:
The inspiration came from grpc (https://godoc.org/google.golang.org/grpc/codes). If you wanted to do something similar with these errors, you could do:

```
switch Code(error) {
// deal with codes
case Unknown:
// not a hashTreeError
}
```

But I can easily make Code an exported field. If you think that's better, let me know and I'll change it.